src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocArguments.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * *
+ *     Copyright 2018 BarD Software s.r.o
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ *     Author: Mikhail Shavkunov (@shavkunov)
+ *
+ */
+
+package com.bardsoftware.backend.fleet.rmarkdown
+
+import org.apache.commons.text.StringSubstitutor
+import java.nio.file.Path
+
+private val pandocArguments = listOf("projectRootAbsPath", "workingDirRelPath",
+        "inputFileName", "outputFileName", "mainFont")
+
+class PandocArguments(
+        projectRootAbsPath: Path,
+        projTasks: Path,
+        mainFile: Path,
+        outputFile: Path,
+        font: String = PANDOC_DEFAULT_FONT) {
+
+    val substitutor: StringSubstitutor
+
+    init {
+        val args = listOf(projectRootAbsPath.toString(), projTasks.toString(),
+                mainFile.toString(), outputFile.toString(), font)
+        val substitutions = (pandocArguments zip args).map { it.first to it.second }.toMap()
+        substitutor = StringSubstitutor(substitutions)
+    }
+
+    fun getCommandLine(compileCommand: String): String {
```

Comment:
It  seems natural to encapsulate access to config into this class as well, no? Because now the knowledge about argument names is here, the knowledge about config key where those arguments are used is in another place.

Reply:
Indeed, done